File: Baekjoon/Java/친구.java
import java.io.*;
import java.util.*;

public class Main {
    static int ans;
    static BufferedReader br;
    static StringTokenizer st;

    public static void main(String[] args) throws IOException {
        br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        String friends[] = new String[n];
        for (int i = 0; i < n; i++) {
            friends[i] = br.readLine();
        }
        ans = 0;
        int[][] result = new int[n][n];

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                for (int k = 0; k < n; k++) {
                    if (i == j) {
                        continue;
                    }

                    if (friends[i].charAt(j) == 'Y') {
                        result[i][j] = 1;
                    } else if (friends[i].charAt(k) == 'Y' && friends[j].charAt(k) == 'Y') {
                        result[i][j] = 1;
                    }
                }
            }
        }

        for (int i = 0; i < n; i++) {
            int sum = 0;
            for (int j = 0; j < n; j++) {
                sum += result[i][j];
            }
            ans = Math.max(ans, sum);
        }
        System.out.println(ans);
    }
}

File: Baekjoon/Python/친구.py
import sys
input = sys.stdin.readline

n = int(input())
friends = [input().rstrip() for _ in range(n)]
result = [[0]*n for _ in range(n)]
ans = 0

for i in range(n):
    for j in range(n):
        for k in range(n): 
            if i == j:
                continue

            if friends[i][j] == 'Y': 
                result[i][j] = 1
            elif (friends[i][k] == 'Y' and friends[k][j] == 'Y'):
                result[i][j] = 1
                
for i in result:
    ans = max(ans,sum(i))

print(ans)
